x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/KeyToSequences.java

```diff
@@ -34,23 +40,42 @@
             this.groups = new SequenceGroup[stages];
         }
 
-        void add(int stage, Sequence sequence) {
+        long add(int stage, Sequence sequence) {
```

Comment:
Accounting/ram usage should not leak into the return values of add/remove/until.